xwiki-pro-macros-ui/src/main/resources/Confluence/Macros/Attachments.xml

```diff
@@ -0,0 +1,1144 @@
+  #end
+#end
+## Filter attachments
+#set ($confluencePatterns = "$!wikimacro.parameters.patterns")
+#if ($confluencePatterns == '')
+  #set($attachmentsFiltered = $attachments)
+#else
+  #set ($patterns = $!confluencePatterns.split(','))
+  #set($attachmentsFiltered = [])
+  #foreach ($attachment in $attachments)
+    #foreach ($pattern in $patterns)
+      #set ($matches = $attachment.filename.matches("(?i)$!pattern"))
+      #if ($matches)
+        #set ($discard = $attachmentsFiltered.add($attachment))
+        #break
+      #end
+    #end
+  #end
+#end
+{{/velocity}}
+{{velocity}}
+## Sort attachments
+#set ($confluenceSortBy = "$!wikimacro.parameters.sortBy")
+#if ($confluenceSortBy == 'date')
+  #set ($sortBy = 'date')
+#elseif ($confluenceSortBy == 'size')
+  #set ($sortBy = 'filesize')
+#elseif ($confluenceSortBy == 'name')
+  #set ($sortBy = 'filename')
+#elseif ($confluenceSortBy == 'creation date')
+  #set ($sortBy = 'date')
+#else
+{{warning}}
+Warning: sortBy parameter must be one of 'date', 'size', or 'name'
+{{/warning}}
+#end
+#set ($confluenceSortOrder = "$!wikimacro.parameters.sortOrder")
+#set ($reverseOrderProperties = ['filesize', 'date'])
+#if ($confluenceSortOrder == 'ascending')
+  #if ($reverseOrderProperties.contains($sortBy))
+    #set ($sortOrder = 'desc')
+  #else
+    #set ($sortOrder = 'asc')
+  #end
+#elseif ($confluenceSortOrder == 'descending')
+  #if ($reverseOrderProperties.contains($sortBy))
+    #set ($sortOrder = 'asc')
+  #else
+    #set ($sortOrder = 'desc')
+  #end
+#else
+{{warning}}
+Warning: sortOrder parameter must be one of 'ascending' or 'descending'
+{{/warning}}
+#end
+#set ($attachmentsFiltered = $collectiontool.sort($attachmentsFiltered, "$sortBy:$sortOrder"))
+## Display attchments
+{{html clean="false"}}
+#if ($attachmentsFiltered.size() &gt; 0)
+  #displayAttachments($attachmentsFiltered)
+  #deleteAttachmentModal()
+#else
+  &lt;p class="noitems"&gt;$services.localization.render('core.viewers.attachments.noAttachments')&lt;/p&gt;
+#end
+{{/html}}
+## Allow upload
+#set ($upload = "$!wikimacro.parameters.upload")
+#if ($upload == 'true' &amp;&amp; ($hasEdit || $hasAdmin) &amp;&amp; $xcontext.action == 'view')
+
+
+  {{html}}
+    &lt;form action="$doc.getURL("upload")" enctype="multipart/form-data" method="post" id="AddAttachment"&gt;
+    &lt;div&gt;
+    ## CSRF prevention
+    &lt;input type="hidden" name="form_token" value="$!{services.csrf.getToken()}" /&gt;
+    ## $redirect is defined in attachmentslist.vm
+    &lt;input type="hidden" name="xredirect" value="$escapetool.xml($redirect)" /&gt;
+    &lt;fieldset id="attachform"&gt;
+      &lt;legend&gt;$services.localization.render('core.viewers.attachments.upload.title')&lt;/legend&gt;
+      ## Temporarily disabled, until we fix attachment name handling
+      ## &lt;div&gt;&lt;label id="xwikiuploadnamelabel" for="xwikiuploadname"&gt;$services.localization.render('core.viewers.attachments.upload.filename')&lt;/label&gt;&lt;/div&gt;
```

Comment:
Missing XML escaping. And line too long (would be good to keep it under 120 characters for improved readability).